tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -274,7 +274,7 @@ uint32_t GetElementCount(clang::QualType type) {
 /// type.</summary>
 QualType GetArrayElementType(clang::QualType type) {
   QualType qTy = type.getCanonicalType().getNonReferenceType();
-  assert(qTy->isArrayType() && "otherwise caller shouldn't be invoking this");
+  DXASSERT_NOMSG(qTy->isArrayType() && "otherwise caller shouldn't be invoking this");
```

Comment:
heh. well, you kind of do have a message, but it ends up being the same anyway. 😄